src/FileWriter.php

```diff
@@ -25,21 +26,25 @@ final class FileWriter
      */
     public static function writeFile($file, $content, $chmod = 0666)
     {
-        $tmp = tempnam(sys_get_temp_dir(), md5($file));
+        $dir = dirname($file);
+        $tmp = tempnam($dir, md5($file));
```

Comment:
Unrelated: The purpose of the prefix is to give the user the ability to identify the source of a temporary file, while `tempnam()` guarantees the name uniqueness regardless of the prefix. Setting it to the hash of a file doesn't seem to provide any value but may do harm.